pkg/agent/utils.go

```diff
@@ -194,6 +195,40 @@ func getBase64EncodedGzippedCustomScript(csFilename string, config *datamodel.No
 	return getBase64EncodedGzippedCustomScriptFromStr(csStr)
 }
 
+func removeComments(b []byte) []byte {
+	var contentWithoutComments []string
+	lines := strings.Split(string(b), "\n")
+	for _, line := range lines {
+		lineNoWhitespace := strings.TrimSpace(line)
+		if isCommentInBeginningOfLine(lineNoWhitespace) {
+			// ignore entire line that is a comment
+			continue
+		}
+		lastHashIndex := strings.LastIndex(line, "#")
+		if lastHashIndex > 0 && isCommentAtTheEndOfLine(lastHashIndex, line) {
+			// remove only the comment part from line
+			line = line[:lastHashIndex]
+		}
+		contentWithoutComments = append(contentWithoutComments, line)
+	}
+	return []byte(strings.Join(contentWithoutComments, "\n"))
+}
+
+// Trying to avoid using a regex. There are certain patterns we ignore just to be on the safe side. This is enough to get rid of most of the obvious comments.
+func isCommentAtTheEndOfLine(lastHashIndex int, trimmedToCheck string) bool {
+	getSlice := func(start, end int, str string) string {
+		if end > len(str) || start > end {
+			return ""
+		}
+		return str[start:end]
+	}
+	return getSlice(lastHashIndex-1, lastHashIndex+1, trimmedToCheck) != "<#" && getSlice(lastHashIndex, lastHashIndex+2, trimmedToCheck) == "# "
+}
+
+func isCommentInBeginningOfLine(trimmedToCheck string) bool {
+	return strings.HasPrefix(trimmedToCheck, "# ") || strings.HasPrefix(trimmedToCheck, "##")
```

Comment:
woudn't this miss comments that start with '#' and immediately proceed with a non-space character?

e.g. "#foo bar baz"

Reply:
it would, but i wasn't aiming to remove every possible comment, this was supposed to be a quick fix :'( but if we want to make it more "thorough" we could try. At that point we might need a regex though?

Reply:
We're gonna have 2 problems now